playbooks/roles/mount-web-directory/tasks/main.yml

```diff
@@ -0,0 +1,10 @@
+---
```

Comment:
I think we don't need a separate role. You can use ansible tags to separate tasks.
Move this task to inside the wordpress role and set a tag like `only-mount`, then run the common server, and then run the wordpress role with the remaining tasks again (maybe using another tag).
Example:
```
---
- name: Deploy the OpenCraft landing page
  hosts: wordpress
  become: true
  vars:
        ....
  roles:
    - role: wordpress
      tags: only-mount

    - common-server
    - geerlingguy.php-versions
    - geerlingguy.php-mysql

    - role: wordpress
      tags: install-deploy

    - tarsnap
```

Reply:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_tags.html

Reply:
Indeed both solutions work. IMHO smaller roles with well-defined scope are easier to read and maintain (for instance, the "wordpress" role should only install the Wordpress content management system and not execute machine-specific tasks, e.g. mount the /var/www directory). Is there any benefit of using tags here besides reducing the number of roles?

Reply:
These playbooks are old and carry a lot of drift with them.

My main concern by splitting the role is that we'd need to add more configuration and secret variables for a proper implementation, passing `/dev/sdX', mount point and permissions of the attached volume from the `ansible-secrets` repo instead of hardcoding them in the role.

I can accept the work as is after you add a README with some context on the role, but it's not the ideal solution.

Reply:
Reading your last comment, I have remembered that the purpose of the Ansible role is to be reusable, so it makes sense. I have changed my patch to use "pre_tasks" instead of a separate role.